src/nsqdconnection.js

```diff
@@ -216,9 +215,14 @@ class NSQDConnection extends EventEmitter {
    * Create a snappy stream.
    */
   startSnappy() {
-    this.inflater = new UnsnappyStream();
-    this.deflater = new SnappyStream();
-    this.reconsumeFrameBuffer();
+    try {
+      const snappystream = require('snappystream');
+      this.inflater = new snappystream.UnsnappyStream();
+      this.deflater = new snappystream.SnappyStream();
+      this.reconsumeFrameBuffer();
+    } catch (_err) {
+      console.warn('optional dependency snappystream could not be loaded')
```

Comment:
You have to throw the error with the new message because, by the point that you get here, the connection will have already told the server that it wants to use Snappy and then parsing errors will result. So, having the warning is a step in the right direction, but failing here is better than failing with a different message and hoping that they spot the warning.